### ISSUE TITLE
Install capstone or ropper install failed

### ISSUE THREAD

**Issue by veritas501** [therealsaumil/armx#20]
emux version: d21ad03

```diff
diff --git a/Dockerfile-emux b/Dockerfile-emux
index 1c348d3..1b61120 100644
--- a/Dockerfile-emux
+++ b/Dockerfile-emux
@@ -31,6 +31,7 @@ RUN apk add \
        socat \
        netcat-openbsd \
        busybox-extras \
+       capstone \
        --no-cache

 # Install Python packages
```

log:

```
$ ./build-emux-docker
[+] Building 116.3s (11/43)
 => [internal] load .dockerignore                                                                 0.0s
 => => transferring context: 109B                                                                 0.0s
 => [internal] load build definition from Dockerfile-emux                                         0.1s
 => => transferring dockerfile: 3.20kB                                                            0.0s
 => [internal] load metadata for docker.io/library/alpine:latest                                  4.1s
 => [internal] load build context                                                                 0.0s
 => => transferring context: 3.79MB                                                               0.0s
 => [1/39] FROM docker.io/library/alpine:latest@sha256:21a3deaa0d32a8057914f36584b5288d2e5ecc984  1.6s
 => => resolve docker.io/library/alpine:latest@sha256:21a3deaa0d32a8057914f36584b5288d2e5ecc9843  0.0s
 => => sha256:21a3deaa0d32a8057914f36584b5288d2e5ecc984380bc0118285c70fa8c9300 1.64kB / 1.64kB    0.0s
 => => sha256:e7d88de73db3d3fd9b2d63aa7f447a10fd0220b7cbf39803c803f2af9ba256b3 528B / 528B        0.0s
 => => sha256:c059bfaa849c4d8e4aecaeb3a10c2d9b3d85f5165c66ad3a4d937758128c4d18 1.47kB / 1.47kB    0.0s
 => => sha256:59bf1c3509f33515622619af21ed55bbe26d24913cedbca106468a5fb37a50c3 2.82MB / 2.82MB    1.2s
 => => extracting sha256:59bf1c3509f33515622619af21ed55bbe26d24913cedbca106468a5fb37a50c3         0.1s
 => [2/39] RUN apk update                                                                         2.2s
 => [3/39] RUN apk add  bash sudo pv tar dialog iptables curl wget tmux git  vim nano asciinema  93.2s
 => [4/39] RUN pip install --upgrade pip                                                          4.8s
 => [5/39] RUN pip install wheel                                                                  2.2s
 => [6/39] RUN pip install cstruct                                                                2.2s
 => ERROR [7/39] RUN pip install ropper                                                           5.8s
------
 > [7/39] RUN pip install ropper:
#10 1.395 Collecting ropper
#10 1.801   Downloading ropper-1.13.6.tar.gz (71 kB)
#10 2.035   Preparing metadata (setup.py): started
#10 2.248   Preparing metadata (setup.py): finished with status 'done'
#10 2.584 Collecting filebytes>=0.10.0
#10 2.693   Downloading filebytes-0.10.2.tar.gz (20 kB)
#10 2.719   Preparing metadata (setup.py): started
#10 2.925   Preparing metadata (setup.py): finished with status 'done'
#10 3.061 Collecting capstone
#10 3.181   Downloading capstone-4.0.2.tar.gz (2.0 MB)
#10 4.036   Preparing metadata (setup.py): started
#10 4.278   Preparing metadata (setup.py): finished with status 'done'
#10 4.283 Building wheels for collected packages: ropper, filebytes, capstone
#10 4.284   Building wheel for ropper (setup.py): started
#10 4.576   Building wheel for ropper (setup.py): finished with status 'done'
#10 4.577   Created wheel for ropper: filename=ropper-1.13.6-py3-none-any.whl size=99742 sha256=f86b56e139bebc81d6f0b05e2f90d6c78603f38855e027551117b9aab4edc531
#10 4.577   Stored in directory: /root/.cache/pip/wheels/77/a4/5d/a4bc1b653bdcce30a17b5cdda8f19da11444bb8640d03ab678
#10 4.580   Building wheel for filebytes (setup.py): started
#10 4.842   Building wheel for filebytes (setup.py): finished with status 'done'
#10 4.842   Created wheel for filebytes: filename=filebytes-0.10.2-py3-none-any.whl size=27853 sha256=20e412d7bbb7a6e1c6c969b24fb2d736607748e32bd93bfd97b001bd898c4950
#10 4.842   Stored in directory: /root/.cache/pip/wheels/c2/51/58/98925d75705ee4df10da42a098d956183bb70661698fd07753
#10 4.845   Building wheel for capstone (setup.py): started
#10 5.085   Building wheel for capstone (setup.py): finished with status 'error'
#10 5.085   ERROR: Command errored out with exit status 1:
#10 5.085    command: /usr/bin/python3 -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-lf8tm_zk/capstone_06da014948bc46f5807200423450bfcb/setup.py'"'"'; __file__='"'"'/tmp/pip-install-lf8tm_zk/capstone_06da014948bc46f5807200423450bfcb/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d /tmp/pip-wheel-nlsbbm44
#10 5.085        cwd: /tmp/pip-install-lf8tm_zk/capstone_06da014948bc46f5807200423450bfcb/
#10 5.085   Complete output (14 lines):
#10 5.085   running bdist_wheel
#10 5.085   running build
#10 5.085   Building C extensions
#10 5.085   make: uname: Operation not permitted
#10 5.085   make: uname: Operation not permitted
#10 5.085   make: /bin/sh: Operation not permitted
#10 5.085   make: /bin/sh: Operation not permitted
#10 5.085   make: /bin/sh: Operation not permitted
#10 5.085   make: /bin/sh: Operation not permitted
#10 5.085   make: /bin/sh: Operation not permitted
#10 5.085   make: /bin/sh: Operation not permitted
#10 5.085   make: mkdir: Operation not permitted
#10 5.085   make: *** [Makefile:492: cs.o] Error 127
#10 5.085   error: [Errno 2] No such file or directory: 'libcapstone.so.4'
#10 5.085   ----------------------------------------
#10 5.085   ERROR: Failed building wheel for capstone
#10 5.085   Running setup.py clean for capstone
#10 5.308 Successfully built ropper filebytes
#10 5.308 Failed to build capstone
#10 5.358 Installing collected packages: filebytes, capstone, ropper
#10 5.397     Running setup.py install for capstone: started
#10 5.630     Running setup.py install for capstone: finished with status 'error'
#10 5.630     ERROR: Command errored out with exit status 1:
#10 5.630      command: /usr/bin/python3 -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-lf8tm_zk/capstone_06da014948bc46f5807200423450bfcb/setup.py'"'"'; __file__='"'"'/tmp/pip-install-lf8tm_zk/capstone_06da014948bc46f5807200423450bfcb/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /tmp/pip-record-hlsw6813/install-record.txt --single-version-externally-managed --compile --install-headers /usr/include/python3.9/capstone
#10 5.630          cwd: /tmp/pip-install-lf8tm_zk/capstone_06da014948bc46f5807200423450bfcb/
#10 5.630     Complete output (14 lines):
#10 5.630     running install
#10 5.630     running build
#10 5.630     Building C extensions
#10 5.630     make: uname: Operation not permitted
#10 5.630     make: uname: Operation not permitted
#10 5.630     make: /bin/sh: Operation not permitted
#10 5.630     make: /bin/sh: Operation not permitted
#10 5.630     make: /bin/sh: Operation not permitted
#10 5.630     make: /bin/sh: Operation not permitted
#10 5.630     make: /bin/sh: Operation not permitted
#10 5.630     make: /bin/sh: Operation not permitted
#10 5.630     make: mkdir: Operation not permitted
#10 5.630     make: *** [Makefile:492: cs.o] Error 127
#10 5.630     error: [Errno 2] No such file or directory: 'libcapstone.so.4'
#10 5.630     ----------------------------------------
#10 5.630 ERROR: Command errored out with exit status 1: /usr/bin/python3 -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-lf8tm_zk/capstone_06da014948bc46f5807200423450bfcb/setup.py'"'"'; __file__='"'"'/tmp/pip-install-lf8tm_zk/capstone_06da014948bc46f5807200423450bfcb/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /tmp/pip-record-hlsw6813/install-record.txt --single-version-externally-managed --compile --install-headers /usr/include/python3.9/capstone Check the logs for full command output.
------
failed to solve with frontend dockerfile.v0: failed to build LLB: executor failed running [/bin/sh -c pip install ropper]: runc did not terminate sucessfully
```

**Comment by therealsaumil** [comment]
Unable to reproduce the error. Can you please provide steps?

**Comment by veritas501** [comment]
Just using wsl2 and ubuntu 20.04 inside, clone the latest code and run:
```
cd emux
./build-emux-volume
./build-emux-docker
```

, then this happend.

**Comment by veritas501** [comment]
https://asciinema.org/a/9cmQR0Z2D7Rwde9VmzCCCAHtN

**Comment by therealsaumil** [comment]
@veritas501 I tried afresh with an Ubuntu 21.04 system and Docker `20.10.7`. It succeeded. However, I don't think it is the docker version that is making it fail or succeed. It seems to be some artifact left behind in docker caches somewhere. I wiped out all the docker images - did a `docker image prune -a` (very drastic measure) to ensure everything was being built from scratch.

I can see that there is an `RUN apk add capstone` in your `Dockerfile-emux`. That isn't present for mine (or also from the latest commit on Github). I would highly recommend trying it completely afresh and see if it still fails to build.
